### DISCUSSION TITLE
[@types/styled-components] Incompatible with tsconfig "moduleResolution": "Node16"

### DISCUSSION THREAD

**Discussion by pschiffmann** [DefinitelyTyped/DefinitelyTyped#63363] [Issues with a @types package]
I have a file `main.tsx` with this content:

```ts
import styled from "styled-components";

const RedDiv = styled("div")({ backgroundColor: "red" });
```

Running `tsc` on my projects gives me the following error:
```
src/main.tsx:3:16 - error TS2349: This expression is not callable.
  Type 'typeof import("<...>/node_modules/@types/styled-components/index")' has no call signatures.

3 const RedDiv = styled("div")({ backgroundColor: "red" });
                 ~~~~~~
```

My `tsconfig.json` contains this line: `"moduleResolution": "Node16"`. When I switch it to `"moduleResolution": "Node"`, the error disappears and TypeScript is able to understand that `styled()` is callable.

Here is a simple project that demonstrates the issue: https://github.com/pschiffmann/styled-components-moduleresolution

[a comment by an automated account is not shown]

**Comment by tpluscode** [comment]
See https://github.com/microsoft/TypeScript/issues/53349

I just learned that for this, the types package must include `"type": "module"`, matching the upstream
